tensorflow/python/data/experimental/ops/optimization_options.py

```diff
@@ -165,18 +134,46 @@ class OptimizationOptions(options.OptionsBase):
       docstring="Whether to fuse shuffle and repeat transformations. If None, "
       "defaults to True.")
 
+  _autotune_option = options.create_option(
+      name="_autotune_option",
+      ty=autotune_options.AutotuneOptions,
+      docstring="The autotune options associated with the dataset. See "
+      "`tf.data.AutotuneOptions` for more details.",
+      default_factory=autotune_options.AutotuneOptions)
+
+  def _get_autotune_option_compatibility_map(self):
+    """Helper method which returns the backward-compatibility
+    attribute mapping between `experimental_optimization` and
+    `autotune` options.
+    """
+    return {
+        # deprecated `experimental_optimization` : new `_autotune_option`
+        "autotune": "enabled",
+        "autotune_buffers": "experimental_autotune_buffers",
+        "autotune_cpu_budget": "cpu_budget",
+        "autotune_ram_budget": "ram_budget"
+    }
+
+  def __getattr__(self, name):
+    # handle backward compatibility with deprecated options
+    compatibility_map = self._get_autotune_option_compatibility_map()
+    if name in compatibility_map:
+      return getattr(self._autotune_option, compatibility_map[name])
+    else:
+      return getattr(self, name)
+
+  def __setattr__(self, name, value):
+    # handle backward compatibility with deprecated options
```

Comment:
This should log a deprecation warning similar to what we do for ThreadingOptions.